### PR TITLE
Adding examples and documentation for local inference (fixed vulnerability)

### PR DESCRIPTION
**Pull request by cat101**
* Added a check to see if model runner is running
 * Added connectivity check for cloud inference services
 * Updated dependencies to fix certifi vulnerability
 * Cleaned up the container based dev environment (not maintained)

### CODE REVIEW

File: landingai/predict.py [hunk 1]
```diff
@@ -191,14 +212,39 @@ def serialize_rois(rois: List[List[Tuple[int, int]]], mode: str) -> str:
...
+        # Check if the inference server is reachable
+        if check_server_ready:
+            sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+            sock.settimeout(1)
+            result = sock.connect_ex((host, port))
+            if result != 0:
+                raise ConnectionError(
+                    f"Failed to connect to the model server. Please check if the server is running and the connection url ({self._url})."
+                )
+            sock.close()
```
Comment (AsiaCao) [inline landingai/predict.py:241]: how about abstract it out as a function and reuse it for both predictors?
just for keeping it DRY

File: landingai/predict.py [hunk 2]
```diff
@@ -52,7 +55,25 @@ def __init__(
...
+            else:
+                port = socket.getservbyname(parsed_url.scheme)
+            sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+            sock.settimeout(1)
+            result = sock.connect_ex((parsed_url.hostname, port))
+            if result != 0:
+                raise ConnectionError(
+                    f"Failed to connect to the cloud inference service. Check that {Predictor._url} is accesible from this device"
+                )
+            sock.close()
```
Comment (AsiaCao) [inline landingai/predict.py:75]: shall we add a test case for this path?

**Review by AsiaCao — APPROVED** [review]
nice

File: landingai/predict.py (hunk 1 above)
Comment (cat101) [reply landingai/predict.py]: Done.

File: landingai/predict.py (hunk 2 above)
Comment (cat101) [reply landingai/predict.py]: Done

**Pull request merged**